OpenRA.Game/GameRules/WeaponInfo.cs

```diff
@@ -161,5 +162,20 @@ public void Impact(Target target, Actor firedBy, IEnumerable<int> damageModifier
 					a();
 			}
 		}
+
+		public WDist ModifiedRange(IEnumerable<IRangeModifier> modifiers)
+		{
+			return new WDist(OpenRA.Traits.Util.ApplyPercentageModifiers(Range.Length, modifiers.Select(m => m.GetRangeModifier())));
+		}
+
+		public WDist DefaultModifiedRange(IEnumerable<IRangeModifierInfo> modifiers)
+		{
+			return new WDist(OpenRA.Traits.Util.ApplyPercentageModifiers(Range.Length, modifiers.Select(m => m.GetRangeModifierDefault())));
+		}
+
+		public WDist ModifiedRange(Actor self)
```

Comment:
WeaponInfo is intended to be a very low level object, so teaching them about `IRangeModifier` isn't great.